File: app/src/main/java/org/tasks/activities/attribution/AttributionAdapter.java
package org.tasks.activities.attribution;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import androidx.recyclerview.widget.RecyclerView.ViewHolder;
import java.util.List;
import org.tasks.R;

public class AttributionAdapter extends RecyclerView.Adapter<ViewHolder> {

  private final List<AttributionRow> rows;

  AttributionAdapter(List<AttributionRow> rows) {
    this.rows = rows;
  }

  @NonNull
  @Override
  public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
    LayoutInflater inflater = LayoutInflater.from(parent.getContext());
    return viewType == 0
        ? new LicenseHeader(inflater.inflate(R.layout.row_attribution_header, parent, false))
        : new LicenseRow(inflater.inflate(R.layout.row_attribution, parent, false));
  }

  @Override
  public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
    AttributionRow row = rows.get(position);
    if (getItemViewType(position) == 0) {
      ((LicenseHeader) holder).bind(row.getLicense());
    } else {
      ((LicenseRow) holder).bind(row.getCopyrightHolder(), row.getLibraries());
    }
  }

  @Override
  public int getItemViewType(int position) {
    return rows.get(position).isHeader() ? 0 : 1;
  }

  @Override
  public int getItemCount() {
    return rows.size();
  }
}

File: app/src/main/java/org/tasks/activities/attribution/AttributionAdapter.kt
package org.tasks.activities.attribution

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import org.tasks.R

class AttributionAdapter internal constructor(private val rows: List<AttributionRow>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        return if (viewType == 0) {
            LicenseHeader(inflater.inflate(R.layout.row_attribution_header, parent, false))
        } else {
            LicenseRow(inflater.inflate(R.layout.row_attribution, parent, false))
        }
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val row = rows[position]
        if (getItemViewType(position) == 0) {
            (holder as LicenseHeader).bind(row.license)
        } else {
            (holder as LicenseRow).bind(row.copyrightHolder, row.libraries)
        }
    }

    override fun getItemViewType(position: Int) = if (rows[position].isHeader) 0 else 1

    override fun getItemCount() = rows.size
}